src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/snippets/insert_trigger_snippet.ts

```diff
@@ -7,47 +7,257 @@
+    }
+
+    if (!trimmed || trimmed.startsWith('#')) {
+      if (lineIndent > triggersIndent || !trimmed) {
+        lastCommentOrEmptyLine = lineNumber;
+      } else {
+        break;
+      }
+    }
+  }
+
+  return lastCommentOrEmptyLine + 1;
+}
+
+// Algorithm:
+// 1. Check if triggers section exists (even if empty or has empty items)
+// 2. If triggers section exists, find the next line after the last trigger node range
+// 3. If no trigger section found, add triggers: section in the first line of yaml
 export function insertTriggerSnippet(
   model: monaco.editor.ITextModel,
-  yamlDocument: Document,
+  yamlDocument: Document | null,
   triggerType: TriggerType,
   editor?: monaco.editor.IStandaloneCodeEditor
 ) {
-  // find triggers: line number and column number
-  const triggerNodes = getTriggerNodes(yamlDocument);
+  let document: Document;
+  try {
+    document = yamlDocument || parseDocument(model.getValue());
+  } catch (error) {
+    // If YAML is malformed, we can't insert the trigger snippet
+    return;
+  }
+
+  const triggerNodes = getTriggerNodes(document);
   const triggerNode = triggerNodes.find((node) => node.triggerType === triggerType);
-  let insertTriggersSection = false;
-  let insertAtLineNumber = 1;
-  let indentLevel = 0;
   if (triggerNode) {
     // do not override existing trigger
     return;
   }
-  if (triggerNodes.length > 0) {
-    const lastTriggerRange = getMonacoRangeFromYamlNode(
+
+  const triggersPair = getTriggersPair(document);
+  let insertTriggersSection = true;
+  let insertAtLineNumber = 1;
+  let replaceRange: monaco.Range | null = null;
+  let indentLevel = 0;
+
+  if (triggersPair) {
+    insertTriggersSection = false;
+    const { range: triggersKeyRange, indentLevel: expectedIndent } = getTriggersKeyInfo(
       model,
-      triggerNodes[triggerNodes.length - 1].node
+      triggersPair
     );
-    if (lastTriggerRange) {
-      // add a newline after the last trigger
-      insertAtLineNumber = lastTriggerRange.endLineNumber;
-      indentLevel = getIndentLevelFromLineNumber(model, lastTriggerRange.startLineNumber);
+
+    // First, check if there are any empty items to replace
+    const firstEmptyItem = findFirstEmptyItem(model, triggersPair);
+    if (firstEmptyItem) {
+      replaceRange = createReplacementRange(model, firstEmptyItem.lineNumber);
+      indentLevel = firstEmptyItem.indentLevel;
+    } else if (triggerNodes.length > 0) {
+      const lastTriggerRange = getMonacoRangeFromYamlNode(
+        model,
+        triggerNodes[triggerNodes.length - 1].node
+      );
+      if (lastTriggerRange) {
+        insertAtLineNumber = lastTriggerRange.endLineNumber;
+        indentLevel = getIndentLevelFromLineNumber(model, lastTriggerRange.startLineNumber);
+      }
+    } else if (triggersKeyRange) {
+      // Triggers section exists but is completely empty, insert after triggers:
+      insertAtLineNumber = getInsertLineAfterTriggersKey(model, triggersKeyRange, expectedIndent);
```

Comment:
can we just use `triggersKeyRange.endLineNumber + 1` instead of the looping in getInsertLineAfterTriggersKey?

Reply:
Addressed in [bbde1aa](https://github.com/elastic/kibana/pull/245875/commits/bbde1aa6a7de938d7d9c67a8c25fc510c191d922).